csaf/src/model/store.rs

```diff
@@ -1,10 +1,9 @@
-use crate::model::metadata::Distribution;
 use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
 use std::path::{Path, PathBuf};
+use url::Url;
 
 /// create a distribution base directory
-pub fn distribution_base(base: impl AsRef<Path>, distribution: &Distribution) -> PathBuf {
-    base.as_ref().join(
-        utf8_percent_encode(distribution.directory_url.as_str(), NON_ALPHANUMERIC).to_string(),
-    )
+pub fn distribution_base(base: impl AsRef<Path>, url: Url) -> PathBuf {
```

Comment:
if we only need a `&str`, use use that type, instead of `Url`, which requires also to be cloned first.